pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/functioncache/FunctionCacheManagerImpl.java

```diff
@@ -74,9 +76,9 @@ public void registerFunctionInstance(String fid,
         }
 
         synchronized (cacheFunctions) {
-            FunctionCacheEntry entry = cacheFunctions.get(fid);
-
-            if (null == entry) {
+            final AtomicBoolean computed = new AtomicBoolean(false);
+            final AtomicReference<Throwable> throwable = new AtomicReference<>();
+            FunctionCacheEntry entry = cacheFunctions.computeIfAbsent(fid, __ -> {
```

Comment:
Please note that the contract of computeIfAbsent is that the function must be side effect free because it can be executed multiple times.
I am not sure that this function is side effect free
Can you please double check?